src/t8_forest/t8_forest.c

```diff
@@ -239,6 +240,26 @@ t8_forest_set_balance (t8_forest_t forest, const t8_forest_t set_from,
   }
 }
 
+void
+t8_forest_set_subelements (t8_forest_t forest, const t8_forest_t set_from)
+{
+  T8_ASSERT (t8_forest_is_initialized (forest));
+
+  if (set_from != NULL) {
+    /* If set_from = NULL, we assume a previous forest_from was set */
+    forest->set_from = set_from;
+  }
+
+  /* Add SUBELEMENTS to the from_method.
+   * This overwrites T8_FOREST_FROM_COPY */
+  if (forest->from_method == T8_FOREST_FROM_LAST) {
+    forest->from_method = T8_FOREST_FROM_SUBELEMENTS;
+  }
+  else {
+    forest->from_method |= T8_FOREST_FROM_SUBELEMENTS;
+  }
```

Comment:
This is a good question. In my opinion the answer depends on the implementation and the usecase of subelements. 

In terms of removing hanging nodes it should not be allowed to use subelements without balancing since we will not be able to remove all hanging nodes from an unbalanced mesh. Furthermore, we use the "t8_forest_leaf_face_neighbors" function which only works for balanced forests (I think). 

Now imagine using subelements for other reasons than hanging nodes (just replacing every element by some children subelements in the end for fun). In this case I see no reason for not allowing them on unbalanced meshs.

Reply:
Maybe it would make sense to give the set_subelements an additional flag parameter (remove hanging nodes yes or no). If yes, we want to remove hanging nodes and therefore the forest must be balanced. So in this case it would be reasonable to set balance there as well. 
If no, then we use subelements for some other reason an do not rely on an balanced forest. In this case we would use some other refine callback function leading to some other subelement types (with other values than 2 to 65 which are reserved for the hanging-nodes-removing subelements).

Reply:
Yes, this would be a good solution. 
What do you think about the idea of an additional parameter for the set_subelement function, clearing the use case of subelements (removing subelements or something else)? I see some nice advantages: 

- the name set_subelements is more reasonable since all usecases of subelements will entry this function and the name is nice, since it continues set_adapt, set_balance etc. 
- within the t8_forest_subelements function we could use this parameter to differ between different callback functions (one for removing hanging nodes and one for some other use case). The callback functions for other usecases could stay unimplemented or be something trivial (like refining all elements by some subelement at the end) and it would be easy to change this callback for whatever reason someone would like to use them in the future. 
- Additionally, we could always start the subelement type at 2 and use the parameter to differ between subelement usecases within the quad_w_sub scheme. The scheme would then get a tupel (subelement type, usecase) instead of just the subelement type. Then there would be no risk of overlapping subelement types for different usecases. 

So, what do you think? At least, I would keep this idea in mind until everything else, important for my thesis, works.

Reply:
For now, I just renamed the set_subelement function to set_remove_hanging_faces and within this function, set_balance is set if that has not happened yet.